### PR TITLE
[Python(pandas)] Scan multiple chunks worth of values from a 'object' dtype DataFrame

### LINKED ISSUE #4689: Regression in distinct query over Python dates in 0.5.0
### What happens?

We are encountering a regression in the behavior of a select distinct query over a dataframe with Python dates in it in 0.5.0

### To Reproduce

I do not have as minimal of a reproduction as with #4688, as this one seems like it may depend on the vast majority of values in the column being the same. But I do have a reproduction. I'm not sure the best way to share a small (under 1mb) parquet file like this these days. [Here's a link](https://www.dropbox.com/s/oijokech2hx1j2w/distinct_dates.parquet?dl=0) from my dropbox that hopefully works, but let me know if there's a better way to share this.

This parquet file has about 850k rows with a single `datetime64[ns]` column named `EVENT_TIME`. The vast majority of rows have the value `2021-01-27T00:00:00`, but not all as we'll see.

```python
import duckdb
import pandas as pd

input_df = pd.read_parquet("distinct_dates.parquet", engine="fastparquet")
conn = duckdb.connect(database= ":memory", read_only=False)
conn.register("input_df", input_df)
conn.execute("select distinct EVENT_TIME from input_df")
initial_result = conn.df()
```

If you inspect `initial_result` you will see that it has 159 rows, all values at midnight on different dates. Now we'll convert these to Python dates and run the same query:

```python
date_df = input_df.copy()
date_df['EVENT_TIME'] = pd.to_datetime(date_df['EVENT_TIME']).dt.date

conn.register("date_df", date_df)
conn.execute("select distinct EVENT_TIME from date_df")
date_result = conn.df()
```

If you inspect `date_result`, it only has a single row, with the value `datetime.date(2021, 1, 27)`. You can double check that `date_df` does contain other values. e.g. `print(date_df['EVENT_TIME'][4496])` yields `datetime.date(2021, 1, 28)`, but they are not showing up in the distinct query. Running the same query on 0.4.0 yields the correct 159 rows with the same input dataframe. I have checked with the latest master branch build and it also appears to be affected by this bug.

### OS:

macOS, Linux

### DuckDB Version:

0.5.0

### DuckDB Client:

Python

### Full Name:

Dylan Scott

### Affiliation:

Hex Technologies

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

### PR DESCRIPTION
**Pull request by Tishj**
This PR fixes #4689 

When writing the conversion for object datatypes I seem to have completely disregarded the offset, so if a DataFrame contains more than STANDARD_VECTOR_SIZE (1024) entries, the same 1024 entries would get scanned over and over again.

### CODE REVIEW

File: tools/pythonpkg/src/python_conversion.cpp [hunk 1]
```diff
@@ -201,59 +201,122 @@ bool TryTransformPythonNumeric(Value &res, py::handle ele) {
...
+		return PythonObjectType::Other;
+	}
+}
+
+Value TransformPythonValue(py::handle ele, const LogicalType &target_type, bool nan_as_null) {
+	auto &import_cache = *DuckDBPyConnection::ImportCache();
+
+	auto object_type = GetObjectType(ele);
+	if (object_type != PythonObjectType::Date) {
+		;
```
Comment (Mytherin) [inline tools/pythonpkg/src/python_conversion.cpp:251]: Is this supposed to be an empty statement?

**Review by Mytherin — COMMENTED** [review]
Thanks for the updates, LGTM. One minor comment:

File: tools/pythonpkg/src/python_conversion.cpp (hunk 1 above)
Comment (Tishj) [reply tools/pythonpkg/src/python_conversion.cpp:251]: Ah crap, that's a leftover from debugging :sweat_smile:

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**